vmm/src/lib.rs

```diff
@@ -915,10 +918,19 @@ impl Vmm {
 
         let mut vcpus = Vec::with_capacity(vcpu_count as usize);
 
-        for cpu_id in 0..vcpu_count {
+        for cpu_index in 0..vcpu_count {
+            #[cfg(target_arch = "x86_64")]
             let io_bus = self.pio_device_manager.io_bus.clone();
-            let mut vcpu = Vcpu::new(cpu_id, &self.vm, io_bus, request_ts.clone())
-                .map_err(StartMicrovmError::Vcpu)?;
+            let mut vcpu = Vcpu::new(
+                cpu_index,
+                self.vm.fd(),
+                #[cfg(target_arch = "x86_64")]
+                self.vm.supported_cpuid().clone(),
+                #[cfg(target_arch = "x86_64")]
+                io_bus,
+                request_ts.clone(),
+            )
```

Comment:
I do not really like functions with variable parameter count based on a compile time switch. It is possible to have 2 separate new functions : new_x86, new_aarch64 or some other way of fixing the variable parameter count.

Reply:
Fine by me, will change unless someone disagrees.

Reply:
@serban300 since you are already reviewing this PR, please comment on this.

Reply:
I agree. Different constructors for x86 vs aarch64 sound good.

Reply:
Instead of having new_x86 and new_aarch64 you can have two new functions compiled only for the targeted platforms.

Reply:
@andreeaflorescu the original/current version of the code in this PR does that. It provides two `Vcpu::new()` functions with different signatures, each compiled for its respective platform.